backend/server.py

```
@@ -47,6 +47,18 @@ def _is_writable(stream):
     print(f"voicebox-server {__version__}")
     sys.exit(0)
 
+# Detect backend variant from binary name BEFORE importing backend modules
+# so that env-var guards in app.py (e.g. HSA_OVERRIDE_GFX_VERSION) fire at import time.
+import re as _re
```

Ruff (0.15.10)

[error] Module level import not at top of file

(E402)